dask_expr/_expr.py

```diff
@@ -3766,19 +3766,44 @@ def _broadcast_dep(self, dep: Expr):
         return dep.npartitions == 1
 
     def _task(self, name: Key, index: int) -> Task:
-        subgraphs = {}
+        internal_tasks = []
+        seen_keys = set()
+        external_deps = set()
         for _expr in self.exprs:
             if self._broadcast_dep(_expr):
                 subname = (_expr._name, 0)
             else:
                 subname = (_expr._name, index)
-            subgraphs[subname] = _expr._task(subname, subname[1])
-
-        for i, dep in enumerate(self.dependencies()):
-            subgraphs[self._blockwise_arg(dep, index)] = "_" + str(i)
+            t = _expr._task(subname, subname[1])
+            assert t.key == subname
+            internal_tasks.append(t)
+            seen_keys.add(subname)
+            external_deps.update(t.dependencies)
+        external_deps -= seen_keys
+        dependencies = {
+            dep: TaskRef(dep)
+            # Note: the method dependencies isn't strictly needed here. We could
+            # also get the list of external dependencies when iterating over the
+            # subgraphs above.
+            for dep in external_deps
+        }
+        t = Task(
+            name,
+            Fused._execute_internal_graph,
+            # Wrap the actual subgraph as a data node such that the tasks are
+            # not erroneously parsed. The external task would otherwise carry
+            # the internal keys as dependencies which is satisfiable
+            DataNode(None, internal_tasks),
+            dependencies,
+            (self.exprs[0]._name, index),
+        )
+        return t
```

Comment:
I implemented this (and similar patterns) now two or three times. I believe I'll factor this out to a classmethod like `Task.fuse_tasks`. **If** we end up rewriting the low level fusion this will obviously also come in handy. For now, we can keep this logic to get out a fix quickly